File: Golang/Easy/1437_Check_if_All_1_Are_at_Least_Length_K_Places_Away/solution.go
package checkifall1areatleastlengthkplacesaway

func kLengthApart(nums []int, k int) bool {
	count := 0
	hasStarted := false

	for i, _ := range nums {
		if nums[i] == 0 {
			count += 1
		} else {
			if count < k && hasStarted {
				return false
			}
			hasStarted = true
			count = 0
		}
	}
	return true
}

File: Python/Easy/1437_Check_if_All_1_Are_at_Least_Length_K_Places_Away/solution.py
from typing import List

class Solution:
    def kLengthApart(self, nums: List[int], k: int) -> bool:
        count = 0
        hasStarted = False
        for i in range(len(nums)):
            if nums[i] == 0:
                count += 1
            else:
                if count < k and hasStarted:
                    return False
                hasStarted = True
                count = 0
        return True